homeassistant/components/media_player/gpmdp.py

```diff
@@ -6,42 +6,144 @@
 
 _LOGGER = logging.getLogger(__name__)
 REQUIREMENTS = ['websocket-client==0.37.0']
 SUPPORT_GPMDP = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK
+GPMDP_CONFIG_FILE = 'gpmpd.conf'
+_CONFIGURING = {}
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
-    """Setup the GPMDP platform."""
-    from websocket import create_connection
+def request_configuration(hass, config, websocket, add_devices_callback):
+    """Request configuration steps from the user."""
+    configurator = get_component('configurator')
+    if 'gpmdp' in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING['gpmdp'], "Failed to register, please try again.")
+
+        return
+
+    # pylint: disable=unused-argument
+    def gpmdp_configuration_callback(callback_data):
+        """The actions to do when our configuration callback is called."""
+        while True:
+            msg = json.loads(websocket.recv())
+            if msg['channel'] == 'connect':
+                if msg['payload'] == "CODE_REQUIRED":
+                    websocket.send('{"namespace": "connect",'
+                                   '"method": "connect",'
+                                   '"arguments": ["Home Assistant",'
+                                   ' "' + callback_data.get('pin') + '"]}')
+                    tmpmsg = json.loads(websocket.recv())
+                    if tmpmsg['channel'] == 'time':
+                        _LOGGER.error('Error setting up GPMDP. Please pause'
+                                      'the desktop player and try again.')
+                        break
+                    code = tmpmsg['payload']
+                    if code == 'CODE_REQUIRED':
+                        continue
+                    setup_gpmdp(hass, config, code,
+                                add_devices_callback)
+                    config_from_file(hass.config.path(GPMDP_CONFIG_FILE),
+                                     {"CODE": code})
+                    websocket.send('{"namespace": "connect",'
+                                   '"method": "connect",'
+                                   '"arguments": ["Home Assistant",'
+                                   ' "' + code + '"]}')
+                    websocket.close()
+                    break
+
+    _CONFIGURING['gpmdp'] = configurator.request_config(
+        hass, "GPM Desktop Player", gpmdp_configuration_callback,
+        description=(
+            'Enter the pin that is displayed in the '
+            'Google Play Music Desktop Player.'),
+        submit_caption="Submit",
+        fields=[{'id': 'pin', 'name': 'Pin Code', 'type': 'number'}]
+    )
+
 
+def setup_gpmdp(hass, config, code, add_devices_callback):
+    """Setup gpmdp."""
+    from websocket import create_connection
     name = config.get("name", "GPM Desktop Player")
     address = config.get("address")
+    if not code:
+        websocket = create_connection(("ws://" + address + ":5672"), timeout=1)
```

Comment:
Creating the url should happen based on a constant as it is done twice.
Also, why create the websocket here and not inside `request_configuration`?